core/activity_pub.md

```diff
@@ -0,0 +1,40 @@
+
+
+## Public/Private Key Authentication
+
+ActivityPub does not specify a specific authentication scheme, however most ActivityPub communities use OAuth2.0 or something similar which involves a traditional username/password scheme. Since all of the community state must be public and shareable between hosts, we cannot rely on username/password schemes or any scheme that gives hosts privileged information on a user’s authentication info. Thus, users must authenticate by proving ownership of a secret private key that corresponds to the public key that is publicly registered on a blockchain.
+
+## URI and DID
+
+URIs (Uniform Resource Identifier) are used within Activity Pub to identify different servers. 
+
+Fora will use [decentralized identifiers](https://www.w3.org/TR/did-core/), to identify both users and platforms. These DIDs must be registered on a blockchain that the platform can query identities from. Users will have a DID so they can self-attest to their identity on the blockchain and then use that DID in any Fora community they wish. They can also retain this identity even as they move to different communities. The DID document for each user must contain the current public key they will use sign messages sent from their Fora user account. 
+
+Fora implementations which rely on the public key registered with a DID instead of registering a public key with a governance module, will benefit from dynamic swapping of public keys. 
+
+Meanwhile communities will also have DIDs that identify them to their users. This DID will also be registered on the blockchain as a government. The DID will resolve to the current hosts of the platform. Any governance decisions which affects the set of hosts will update the DID resolving result to output the new set of hosts.
+
+## Server
+
+In addition to maintaining the full state of the community, servers will listen to events on the ActivityPub network to record new interactions. Servers must be connected to the appropriate blockchains or DID resolution services in order to track and verify the DID of users in the community.
+
+## Limitations
+
+Since any content posted to a Fora community must contend with the possibility of byzantine host behaviour, the Fora protocol cannot make any guarantees regarding private posts that are only readable to a subset of users. Thus, Fora will only support Public posts and direct messages. Public posts will be sent in plaintext and signed by the sender, while direct messages to users can be end-to-end encrypted given that all users have a Public/Private key as mentioned above (though a different pair would be used for DMs). This ensures that hosts have no ability to attack the confidentiality or integrity of user-sent messages. They can only refuse to relay the messages, though this is grounds for enforcers to remove a host.
```

Comment:
```suggestion
Since any content posted to a Fora community must contend with the possibility of byzantine host behaviour, the Fora protocol cannot make any guarantees regarding private posts that are only readable to a subset of users. Thus, Fora will only support Public posts and direct messages. Public posts will be sent in plaintext and signed by the sender, while direct messages to users can be end-to-end encrypted given that all users have a Public/Private key as mentioned above (though a different pair would be used for DMs). This ensures that hosts have no ability to attack the confidentiality or integrity of user-sent messages. They can only refuse to relay the messages, though the message may simply get relayed through a different path.
```

There isn't any absolute way to remove a "host", bad hosts must simply be circumvented.